velox/connectors/hive/HiveDataSource.h

```diff
@@ -71,6 +71,19 @@ class HiveDataSource : public DataSource {
 
   int64_t estimatedRowSize() override;
 
+  std::unique_ptr<common::Filter> getFilter(
+      column_index_t outputChannel) override {
+    auto& fieldSpec = scanSpec_->getChildByChannel(outputChannel);
+    return fieldSpec.filter()->clone();
```

Comment:
The filter should be MT safe as far as I see, all the methods should be read only.  Can you try using a `shared_ptr<const Filter>` to avoid deep copy?

Reply:
Yes, you are right, changed.